linker/shared/src/main/scala/org/scalajs/linker/backend/wasmemitter/FunctionEmitter.scala

```diff
@@ -2428,8 +2440,9 @@ private class FunctionEmitter private (
 
             case _ =>
               targetWasmType match {
-                case watpe.RefType(true, watpe.HeapType.Any) =>
-                  () // nothing to do
+                case watpe.RefType(targetNullable, watpe.HeapType.Any) =>
+                  if (!targetNullable)
+                    fb += wa.RefAsNonNull
```

Comment:
Just to check my understanding: The nullability cast is actually new here, correct? Before that, it was folded into a full ref cast in the case `targetWasmType: watpe.RefType`?

If yes, consider factoring into a separate commit, even if a bit pedantic. But if ever anybody will have to look at this later, it's going to be much easier to understand what is going on.